### PR TITLE
Milk Pumping Solution

### PR DESCRIPTION
**Pull request by jessechoe10**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-969.mdx [hunk 1]
```diff
@@ -277,3 +277,170 @@ public class pump{
...
+using ll = long long;
+using vl = vector<ll>;
+using pl = pair<ll, ll>;
+using vpl = vector<pl>;
+
+#define pb push_back
+#define f first
+#define s second
+
+const ll MAXN = 1000, INF = 1e9;
```
Comment (SansPapyrus683) [inline solutions/usaco-969.mdx:311]: don't use INF, use INT32_MAX

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-969.mdx [hunk 2]
```diff
@@ -277,3 +277,157 @@ public class pump{
...
+	for (ll i = 0; i < n; i++)
+	{
+		for (ll j = 0; j <= MAXN; j++)
+		{
+			dp[i][j] = INF;
+		}
+	}
+
+	// Set the base cases of the DP
+	forn(j, MAXN + 1) dp[0][j] = 0;
```
Comment (SansPapyrus683) [inline solutions/usaco-969.mdx:333]: oh no

File: solutions/usaco-969.mdx [hunk 3]
```diff
@@ -277,3 +277,157 @@ public class pump{
...
+			{
+				q.push(u.f);
+			}
+		}
+	}
+	
+	for (ll i = 0; i <= MAXN; i++)
+	{
+		// Check if 10^6 * flow rate - r * cost >= 0
+		ll cost = 1e6 * i - r * dp[n - 1][i];
```
Comment (SansPapyrus683) [inline solutions/usaco-969.mdx:383]: 1e6 magic number

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-969.mdx [hunk 4]
```diff
@@ -277,3 +277,157 @@ public class pump{
...
+ - $\texttt{dp}[i][k] = \min_{1 \leq k \leq j}{(\texttt{dp}[x][k] + \texttt{cost}[i][x])}$, where $x$ represents an arbitrary junction connected to $i$ and $\texttt{cost}[i][x]$ represents the cost of the pipe connecting junction $i$ and junction $x$.
+
+Similar to [Talent Show](http://www.usaco.org/index.php?page=viewproblem2&cpid=839), this problem asks us to find the maximum ratio multiplied by a large constant (in this case $10^6$) rounded down to the nearest integer, so we can binary search on the answer for this ratio!
+
+Let $R$ be the maximum ratio that we're trying to find multiplied by $10^6$. The given ratio $R$ works if $\frac{10^6 \cdot F}{C} \geq R$ is satisfied, where $F$ represents the flow rate in the path, and $C$ represents the total path cost in that path. This condition can more simply be written as $10^6 \cdot F - R \cdot C \geq 0$, where $C$ is $\texttt{dp}[n - 1][j]$ and $F$ is $j$, where the condition $1 \leq j \leq 10^6$ must hold.
```
Comment (bqi343) [inline solutions/usaco-969.mdx:290]: there's no need to binary search on R considering that you only use R at the end.

File: solutions/usaco-969.mdx [hunk 5]
```diff
@@ -277,3 +277,157 @@ public class pump{
...
+
+<LanguageSection>
+
+<CPPSection>
+	
+**Time Complexity:** $\mathcal{O}(N(N \cdot MAXF + M)\log MAXR)$
```
Comment (bqi343) [inline solutions/usaco-969.mdx:296]: this isn't supposed to pass :|

**Review by bqi343 — CHANGES_REQUESTED** [review]

File: solutions/usaco-969.mdx (hunk 4 above)
Comment (jessechoe10) [reply solutions/usaco-969.mdx:290]: It's an option tho

File: solutions/usaco-969.mdx (hunk 5 above)
Comment (jessechoe10) [reply solutions/usaco-969.mdx:296]: My code doesn't pass or i have the wrong time complexity?

File: solutions/usaco-969.mdx (hunk 4 above)
Comment (bqi343) [reply solutions/usaco-969.mdx:290]: there's no benefit to doing so

File: solutions/usaco-969.mdx (hunk 5 above)
Comment (bqi343) [reply solutions/usaco-969.mdx:296]: `N*N*MAXF*log MAXR > 10^9`

**Comment by bqi343** [comment]
Closing since a solution with this time complexity isn't supposed to pass :P

**Pull request closed without merging**